### DISCUSSION TITLE
package failed

### DISCUSSION THREAD

**Discussion by wwzorz** [AvaloniaUI/Avalonia#15754] [Q&A]
dotnet tool install --global dotnet-deb
dotnet deb install
dotnet restore -r linux-x64
dotnet msbuild Detection.Desktop.csproj /t:CreateDeb /p:TargetFramework=net8.0 /p:RuntimeIdentifier=linux-x64 /p:Configuration=Release

error
C:\Users\admin\.nuget\packages\packaging.targets\0.1.226\build\Packaging.Targets.targets(206,5): error MSB4018: “DebTask”任务意外失败。 [D:\CompanyGitLa
b\detect
ion\src\DetectionClient\Client\Detection.Desktop\Detection.Desktop.csproj]
C:\Users\admin\.nuget\packages\packaging.targets\0.1.226\build\Packaging.Targets.targets(206,5): error MSB4018: System.ArgumentException: Specified prese t is not supported [D:\CompanyGitLab\detection\src\DetectionClient\Client\Detection.Desktop\Detection.Desktop.csproj]
C:\Users\admin\.nuget\packages\packaging.targets\0.1.226\build\Packaging.Targets.targets(206,5): error MSB4018:    at Packaging.Targets.IO.XZOutputStream .Write(Byte[] buffer, Int32 offset, Int32 count) [D:\CompanyGitLab\detection\src\DetectionClient\Client\Detection.Desktop\Detection.Desktop.csproj]
C:\Users\admin\.nuget\packages\packaging.targets\0.1.226\build\Packaging.Targets.targets(206,5): error MSB4018:    at System.IO.Stream.CopyTo(Stream dest ination, Int32 bufferSize) [D:\CompanyGitLab\detection\src\DetectionClient\Client\Detection.Desktop\Detection.Desktop.csproj]
C:\Users\admin\.nuget\packages\packaging.targets\0.1.226\build\Packaging.Targets.targets(206,5): error MSB4018:    at Packaging.Targets.DebTask.Execute()  [D:\CompanyGitLab\detection\src\DetectionClient\Client\Detection.Desktop\Detection.Desktop.csproj]
C:\Users\admin\.nuget\packages\packaging.targets\0.1.226\build\Packaging.Targets.targets(206,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecuti onHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [D:\CompanyGitLab\detection\src\DetectionClient\Client\Detection.Desktop\Detection.Desktop.cs proj]
C:\Users\admin\.nuget\packages\packaging.targets\0.1.226\build\Packaging.Targets.targets(206,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder .ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecution Mode howToExecuteTask) [D:\CompanyGitLab\detection\src\DetectionClient\Client\Detection.Desktop\Detection.Desktop.csproj]

How to fixed it,thanks

**Comment by thevortexcloud** [comment]
No idea how that specific tool works, but generally you use Linux not windows to make Linux packages.

**Comment by wwzorz** [comment]
我使用avalonia 编写的客户端程序，想直接打包成deb 安装在国产边操作系统中，麻烦给一个在linux 上打包的示例，多谢

**Reply by rabbitism** [reply]
https://docs.avaloniaui.net/docs/deployment/debian-ubuntu
